lib/avo/services/authorization_service.rb

```diff
@@ -87,13 +95,14 @@ def defined_methods(user, record, **args)
         end
       end
 
-      def initialize(user = nil, record = nil)
+      def initialize(user = nil, record = nil, policy_class: nil)
         @user = user
         @record = record
+        @policy_class = policy_class
```

Comment:
what do you say if we  initialize this this way, `@policy_class = policy_class || Pundit.policy(user, record).class`. I think this can make all other logic simpler

Reply:
this way we can use`policy_class.new`everywhere instead of `if policy_class this else that end`